src/tests/TestServerResolver/TestServerResolver.cpp

```diff
@@ -68,6 +69,52 @@ void TestServerResolver::simpleSrv() {
 	QVERIFY(hasipv4 || hasipv6);
 }
 
+void TestServerResolver::simpleCNAME() {
+	ServerResolver r;
+	QSignalSpy spy(&r, SIGNAL(resolved()));
+
+	QString hostname = QString::fromLatin1("cname.serverresolver.mumble.info");
+	quint16 port = 64738;
+
+	r.resolve(hostname, port);
+
+	// Equivalent to QSignalSpy::wait() in Qt 5.
+	{
```

Comment:
Now that #3170 landed should call `signalSpyWait` (need to rebase onto new master).